modules/python/tests/test_python_options.c

```diff
@@ -219,6 +217,30 @@ Test(python_options, test_python_options)
   }
   Py_DECREF(options_dict);
   PyGILState_Release(gstate);
+}
 
-  python_options_free(options);
+Test(python_options, test_python_options)
+{
+  const gchar *script = "import _syslogng\n"
+                        "assert options['string'] == 'example-value', 'Actual: {}'.format(options['string'])\n"
+                        "assert options['long'] == -42, 'Actual: {}'.format(options['long'])\n"
+                        "assert options['double'] == -13.37, 'Actual: {}'.format(options['double'])\n"
+                        "assert options['boolean'] == True, 'Actual: {}'.format(options['string'])\n"
+                        "assert options['string_list'] == ['example-value-1', 'example-value-2'], "
+                        "'Actual: {}'.format(options['string_list'])\n"
+                        "assert isinstance(options['template'], _syslogng.LogTemplate), 'Actual: {}'.format(type(options['template']))\n"
+                        "assert str(options['template']) == '${example-template}', 'Actual: {}'.format(str(options['template']))\n";
+  PythonOptions *options = _generate_options_test_options();
+
+  _run_py_options_test(options, script);
+
+  gchar *script2 = g_strdup_printf("%s%s", script, script);
+  PythonOptions *options2 = _generate_options_test_options();
+  python_options_add_options(options2, options);
+  python_options_release(options);
+
+  _run_py_options_test(options2, script2);
+
+  g_free(script2);
+  python_options_free(options2);
```

Comment:
Sorry if I am missing something, but I do not how this test case works.

We test an `Options` object with every option types, then we are merging it with the same values and running the test two more times on it?

I think we should test `add_options()` with options that are not already present in the `Options` object. Also I do not understand the reason behind `gchar *script2 = g_strdup_printf("%s%s", script, script);`, it should be enough to run the script once.

Reply:
oh, I see, did not take into account that these are named params
will fix it

Reply:
Ah, I see why you did that :) Yes they are named. Thanks for changing it!

Reply:
hope it's fine now